odoo/service/server.py

```diff
@@ -212,6 +272,51 @@ def signal_handler(self, sig, frame):
             # interrupt run() to start shutdown
             raise KeyboardInterrupt()
 
+    def process_limit(self):
+        rss, vms = memory_info(psutil.Process(os.getpid()))
+        if vms > config['limit_memory_soft']:
+            _logger.info('Thread virtual memory limit (%s) reached.', vms)
+            self.limits_reached_threads.add(threading.currentThread())
+
+        for thread, thread_info in self.threads.items():
```

Comment:
That's O(n**2)

Reply:
wut ?

Reply:
Because you use a property you hide the fact that's it's every access to the property is O(n)

Reply:
Let's talk about it IRL.